apps/examples/smartfs_powercut/smartfs_powercut_main.c

```diff
@@ -171,76 +212,148 @@ static int create_file(char *path, ssize_t bufsize, file_type_t type)
 	return ret;
 }
 
-static int verify_random_file(char *path, int bufsize, int *index)
+static int verify_random_file_data(int fd, int bufsize)
+{
+	int len;
+	uint16_t crc;
+	int data_len = bufsize - sizeof(g_buf.crc);
+
+	len = read(fd, (void *)&g_buf, bufsize);
+	if (len != bufsize) {
+		/* If write is not finished properly */
+		printf("verify_random_file : datachunk is not written fully, ret : %d\n", len);
+		return ERROR;
+	} else {
+		/* Maybe this won't happen but check crc for double confirmation */
+		crc = crc16((const uint8_t *)g_buf.data, data_len);
+		if (crc != g_buf.crc) {
+			printf("verify_random_file : crc mismatch, calulcated crc : %d, written crc : %d\n", crc, g_buf.crc);
+			return ERROR;
+		}
+	}
+	return OK;
+}
+
+static int verify_text_file_data(int fd, char *sample_buf, uint16_t sample_crc, int bufsize)
+{
+	int len;
+	uint16_t crc;
+
+	len = read(fd, (void *)sample_buf, bufsize);
+	if (len != bufsize) {
+		/* If write is not finished properly */
+		printf("verify_text_file : datachunk is not written fully, ret : %d\n", len);
+		return ERROR;
+	} else {
+		/* Maybe this won't happen but check crc for double confirmation */
+		crc = crc16((const uint8_t *)sample_buf, bufsize);
+		if (crc != sample_crc) {
+			printf("verify_random_file : crc mismatch, calulcated crc : %d, Sample File crc : %d\n", crc, sample_crc);
+			return ERROR;
+		}
+	}
+	return OK;
+}
+
+static int verify_file(char *path, int bufsize, int *index, file_type_t type)
 {
 	int fd;
+	int fd_sample;
 	int ret;
-	ssize_t len;
 	off_t size;
 	int i;
-	uint16_t crc;
 	struct stat st;
-	int data_len = bufsize - sizeof(g_buf.crc);
+	uint16_t sample_crc = 0;
+	char *tmp_buf = NULL;
 	
 	fd = open(path, O_RDWR);
 	if (fd < 0) {
 		if (errno == ENOENT) {
-			printf("Verify_file File is not exist, no need to verify!! path : %s\n", path);
+			printf("File to be verified does not exist, no need to verify!! path : %s\n", path);
 			*index = FILE_CHUNK_NO_INDEX;
 			return OK;
 		}
 		
-		printf("verify_file %s open error... errno : %d\n", path, errno);
+		printf("Unable to open file : %s for verifying, errno : %d\n", path, errno);
 		return ERROR;
 	}
 
+	if (type == FILE_TYPE_TEXT) {
+		fd_sample = open(SAMPLE_TXT_FILE, O_RDOK);
+		if (fd_sample < 0) {
+			printf("Unable to open Sample Text File : %s, errno : %d\n", SAMPLE_TXT_FILE, errno);
+			ret = ERROR;
+			goto error_with_fd;
+		}
+
+		tmp_buf = (char *)kmm_malloc(bufsize);
```

Comment:
you can use g_buf again, you do not handle something with 2 buffers...

Reply:
Yes I tried that, but faced an issue

Line 305 `sample_crc = crc16((const uint8_t *)tmp_buf, bufsize);`
I can't calculate crc16 on g_buf it seems because g_buf is a struct with 2 members of different types

So pointer typecasting gave me an error when I used the following: -
`sample_crc = crc16((const uint8_t *)&g_buf, bufsize);`

Reply:
That is pointer, you can use g_buf.data but length is bufsize.